src/types/timestamp.rs

```diff
@@ -1200,8 +1162,7 @@ impl TryInto<ion_c_sys::timestamp::IonDateTime> for Timestamp {
             (Precision::Month, _) => TSPrecision::Month,
             (Precision::Day, _) => TSPrecision::Day,
             (Precision::HourAndMinute, _) => TSPrecision::Minute,
-            (Precision::Second, None) => TSPrecision::Second,
```

Comment:
Note that we are losing the ability to convert into a `TSPrecision::Second`.

Should we retain that mapping by doing a `is_zero()` check on the `mantissa`?

Reply:
Ah looks like we have a test for this exact scenario: `types::timestamp::ionc_tests::convert_from_to_ionc::case_6_second_plus0400`

Reply:
The `ion-c` bindings have been removed since the last release, which I think obviates this issue. Could you try merging from `main` when you have a chance?